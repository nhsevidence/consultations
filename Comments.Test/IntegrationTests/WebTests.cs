using Comments.Test.Infrastructure;
using System.Threading.Tasks;
using Xunit;

namespace Comments.Test.IntegrationTests
{
    public class WebTests : TestBase
    {
        [Fact]
        public async Task Get_Consultations_Homepage()
        {
            // Act
            var response = await _client.GetAsync("/consultations"); //this route is temporary.
            response.EnsureSuccessStatusCode();

            var responseString = await response.Content.ReadAsStringAsync();

            // Assert
            responseString.ShouldMatchApproved();
        }

        [Fact]
        public async Task Get_React_Document_Page()
        {
            // Act
            var response = await _client.GetAsync("/consultations/1/1/introduction");
            response.EnsureSuccessStatusCode();

            var responseString = await response.Content.ReadAsStringAsync();

            // Assert
            responseString.ShouldMatchApproved();
        }
    }
}
