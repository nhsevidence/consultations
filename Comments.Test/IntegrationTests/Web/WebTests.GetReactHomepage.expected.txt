
<!DOCTYPE html>
  <html class="no-js" lang="en-GB">
    <head>
      <meta charset="utf-8">
        <meta content="IE=edge,chrome=1" name="X-UA-Compatible">
          <meta name="viewport" content="width=device-width,initial-scale=1,shrink-to-fit=no">
            <base href="/consultations/"/>
              <meta name="theme-color" content="#004650">
                <script src="/vendor/modernizr-custom.js">
                </script>
                <link rel="manifest" href="/manifest.json">
                  <meta name="msapplication-TileColor" content="#f6f6f6"/>
                  <meta name="msapplication-TileImage" content="//cdn.nice.org.uk/V3/Content/nice/favicon-144.png"/>
                    <link rel="shortcut icon" href="//cdn.nice.org.uk/V3/Content/nice/favicon.ico"/>
                      <link rel="apple-touch-icon-precomposed" href="//cdn.nice.org.uk/V3/Content/nice/favicon-152.png">
                        <link rel="apple-touch-icon" href="//cdn.nice.org.uk/V3/Content/nice/favicon-152.png">
                          <link rel="apple-touch-icon-precomposed" sizes="120x120" href="//cdn.nice.org.uk/V3/Content/nice/favicon-120.png">
                            <link rel="apple-touch-icon" sizes="120x120" href="//cdn.nice.org.uk/V3/Content/nice/favicon-120.png">
                              <link rel="apple-touch-icon-precomposed" sizes="152x152" href="//cdn.nice.org.uk/V3/Content/nice/favicon-152.png">
                                <link rel="apple-touch-icon" sizes="152x152" href="//cdn.nice.org.uk/V3/Content/nice/favicon-152.png">
                                  <link href="//fonts.googleapis.com/css?family=Lato:400,700,900" rel="stylesheet"/>
                                    <link rel="stylesheet" href="/main.css">
                                      <!--[if lt IE 9]>
                                        <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv.min.js">
                                        </script>
                                        <script src="//cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.7/es5-shim.min.js">
                                        </script>
                                        <script src="//cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.10/es5-sham.min.js">
                                        </script>
                                        <script src="//cdnjs.cloudflare.com/ajax/libs/respond.js/1.4.2/respond.min.js">
                                        </script>
                                        <![endif]-->
                                        </head>
                                        <body>
                                          <div id="root">
                                            <div data-reactroot="">
                                              <div class="container">
                                                <div class="grid">
                                                  <div data-g="12">
                                                    <p class="phase-banner">
                                                      <span class="phase-banner__tag">
                                                        <span class="tag tag--impact tag--alpha">
                                                          Alpha
                                                        </span>
                                                      </span>
                                                      <span class="phase-banner__label">
                                                        NICE Consulations is in development. This means it is not feature complete and there may be issues. Find any? 
                                                        <a href="https://github.com/nhsevidence/consultations" rel="noopener noreferrer" target="_blank">
                                                          Please, let us know!
                                                        </a>
                                                      </span>
                                                    </p>
                                                  </div>
                                                  <div data-g="12">
                                                    <div>
                                                      <nav aria-label="Breadcrumbs">
                                                        <p class="visually-hidden">
                                                          You are here:
                                                        </p>
                                                        <ol class="breadcrumbs" aria-labelledby="breadcrumb-label" itemscope="" itemType="http://schema.org/BreadcrumbList">
                                                          <li class="breadcrumbs__crumb" itemProp="itemListElement" itemscope="" itemType="http://schema.org/ListItem">
                                                            <a href="/document" itemProp="name">
                                                              Home
                                                            </a>
                                                            <meta itemProp="position" content="0"/>
                                                          </li>
                                                          <li class="breadcrumbs__crumb" itemProp="itemListElement" itemscope="" itemType="http://schema.org/ListItem">
                                                            <a itemProp="name">
                                                              NICE Guidance
                                                            </a>
                                                            <meta itemProp="position" content="1"/>
                                                          </li>
                                                          <li class="breadcrumbs__crumb" itemProp="itemListElement" itemscope="" itemType="http://schema.org/ListItem">
                                                            <a itemProp="name">
                                                              In Consulation
                                                            </a>
                                                            <meta itemProp="position" content="2"/>
                                                          </li>
                                                          <li class="breadcrumbs__crumb" itemProp="itemListElement" itemscope="" itemType="http://schema.org/ListItem">
                                                            <a itemProp="name">
                                                              Document title
                                                            </a>
                                                            <meta itemProp="position" content="3"/>
                                                          </li>
                                                        </ol>
                                                      </nav>
                                                      <div class="page-header">
                                                        <h1 class="page-header__heading">
                                                          Intravenous fluid therapy in children and young people in hospital : Consultation
                                                        </h1>
                                                        <p class="page-header__lead">
                                                          In development [NG29] Expected publication date TBC
                                                        </p>
                                                      </div>
                                                      <div class="grid">
                                                        <div data-g="12 md:3">
                                                          <nav class="stacked-nav" aria-label="{root.label}">
                                                            <h2 class="stacked-nav__root">
                                                              <a href="#">
                                                                Chapters in this document
                                                              </a>
                                                            </h2>
                                                            <ul class="stacked-nav__list">
                                                              <li class="stacked-nav__list-item">
                                                                <a>
                                                                  Key priorities for implementation
                                                                </a>
                                                              </li>
                                                              <li class="stacked-nav__list-item">
                                                                <a>
                                                                  Recommendations
                                                                </a>
                                                              </li>
                                                              <li class="stacked-nav__list-item">
                                                                <a>
                                                                  Intravenous fluid therapy in children and young people in hospital
                                                                </a>
                                                              </li>
                                                              <li class="stacked-nav__list-item">
                                                                <a>
                                                                  Context
                                                                </a>
                                                              </li>
                                                              <li class="stacked-nav__list-item">
                                                                <a>
                                                                  Recommendations for research
                                                                </a>
                                                              </li>
                                                            </ul>
                                                          </nav>
                                                          <nav class="stacked-nav" aria-label="{root.label}">
                                                            <h2 class="stacked-nav__root">
                                                              <a href="#">
                                                                Additional documents to comment on
                                                              </a>
                                                            </h2>
                                                            <ul class="stacked-nav__list">
                                                              <li class="stacked-nav__list-item">
                                                                <a>
                                                                  Intravenous fluid therapy in children and young people in hospital - Short Guideline
                                                                </a>
                                                              </li>
                                                            </ul>
                                                          </nav>
                                                        </div>
                                                        <div data-g="12 md:6">
                                                          <div class="document-comment-container">
                                                            <div>
                                                            </div>
                                                          </div>
                                                        </div>
                                                        <div data-g="12 md:3">
                                                          <nav class="in-page-nav" aria-labelledby="inpagenav-title">
                                                            <h2 id="inpagenav-title" class="in-page-nav__title">
                                                              On this page
                                                            </h2>
                                                          </nav>
                                                        </div>
                                                      </div>
                                                    </div>
                                                  </div>
                                                </div>
                                              </div>
                                            </div>
                                          </div>
                                          <script src="//cdn.nice.org.uk/V3.1/Scripts/nice/NICE.TopHat.min.js">
                                          </script>
                                          <script type="text/javascript" src="/static/js/main.js">
                                          </script>
                                        </body>
                                      </html>